### PR TITLE
App: Add sourcegraph:// url scheme handler

### LINKED ISSUE #51270: app: Handle sourcegraph:// scheme deep link urls
Allow App to open any link (whether you open it from a browser or anywhere) that starts with `sourcegraph://`

The path following the `sourcegraph://` scheme in the url should be opened inside of the App. For example `sourcegraph://settings` opens `http://localhost:3080/settings` inside of the Tauri webview.

App should launch if it's not already running.

App should gain focus if it's already running.

### PR DESCRIPTION
**Pull request by marekweb**
Add the ability for App to register a handler for scheme urls that start with `sourcegraph://`

The url can then be used to open a specific page in the App (also known as deep linking).

If the app isn't already running, opening a `sourcegraph://` url will launch it. The url's path will be opened inside of the app's webview.

The implementation relies on an entry in `Info.plist` on mac, and for that reason it doesn't work using `sg start dev`; it works when doing a release build.

Tested on Mac so far. There may be tweaks needed for non-mac for initial launch, I'll test on Linux when I get a chance and that may be a followup PR.

Closes #51270

## Test plan

- Run build-release.sh
- Open a link like sourcegraph://settings from anywhere

### CODE REVIEW

File: cmd/frontend/internal/auth/userpasswd/app.go [hunk 1]
```diff
@@ -109,11 +109,19 @@ func AppSignInMiddleware(db database.DB, handler func(w http.ResponseWriter, r *
 			return errors.Wrap(err, "Could not create new user session")
 		}
 
-		// Success. Redirect to search
-		url := r.URL
-		url.RawQuery = ""
-		url.Path = "/search"
-		http.Redirect(w, r, url.String(), http.StatusTemporaryRedirect)
+		// Success. Redirect to search or to returnTo if present.
```
Comment (marekweb) [inline cmd/frontend/internal/auth/userpasswd/app.go:112]:
```suggestion
		// Success. Redirect to search or to redirect if present.
```

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -38,10 +74,40 @@ fn main() {
...
+        .setup(|app| {
             start_embedded_services();
+
+            // Register handler for sourcegraph:// scheme urls.
+            let handle = app.handle();
+            tauri_plugin_deep_link::register(SCHEME, move |request| {
+                let path: &str = extract_path_from_scheme_url(&request);
+
+                // Case 1: the app has been *launched* with the scheme
+                // url. In the frontend, app-main.tsx will read it with
```
Comment (marekweb) [inline src-tauri/src/main.rs:86]:
```suggestion
                // url. In the frontend, app-shell.tsx will read it with
```

**Review by chwarwick — APPROVED** [review]
This makes sense to me.  Tricky with every file basically being a different language

**Comment by vovakulikov** [comment]
@marekweb I see that at the moment release build pipette doesn't work because of Bazel problem, is there a way to test this?

**Pull request merged**